Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -425,15 +425,15 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
 
     cookie.created = WallTime::now().secondsSinceEpoch().milliseconds();
 
-    cookie.domain = options.domain.isNull() ? domain : WTFMove(options.domain);
+    cookie.domain = options.domain.isNull() ? domain : options.domain;
```

Comment:
I think you don't need to set cookie.domain if options.domain is null. If domain is not set, cookie is host-only (so no need to specify domain):
```
cookie.domain = options.domain;
if (!cookie.domain.isNull()) {
    ....
    // In CFNetwork, a domain without a leading dot means host-only cookie.
    // If a non-null domain was passed in, prepend dot to domain to set
    // host-only to false and make the cookie accessible by subdomains.
    cookie.domain = makeString('.', cookie.domain);
}

```

Reply:
If we don't set cookie.domain, it will be null.

But the spec says that the if no domain is passed in, it should default to the current document/service worker's domain:
https://wicg.github.io/cookie-store/#CookieStore-set

Reply:
I see; yes then let's do the original way.